Simulation-Software-Engineering/cpack-exercise-wt2425
# Add installation and packaging targets [mullamm]

GitLab : mullamm
To Test the Code:
	1. Build the Docker image: docker build -t <image-name> .
	2. Run the Docker container: docker run --rm -it --mount type=bind,source="$(pwd)",target=/mnt/cpack-exercise <image-name>
	3. Inside the Docker container: 
	 mkdir build 
cd build
cmake ..
make
make package

	4.	Verify package creation: The .deb and .tar.gz packages should be generated inside the build directory. Inspect the contents of both the .deb and .tar.gz files to ensure they contain the correct files and follow the Debian packaging conventions.
	5. Lintian check:
 
<img width="705" alt="1" src="https://github.com/user-attachments/assets/caf242fc-0546-4d13-ad89-26d5e8428089">
1. “extended-description-is-empty” : CPACK_PACKAGE_DESCRIPTION is set
6. “no-changelog” : Create a file changelog (in cmake/debian/). Install the changelog file in the CMake configuration.
7. “no-copyright-file” : Create a copyright file (in cmake/debian/). Install the copyright file in the CMake configuration.
8. “file-in-unusual-dir” : Installation paths corrected in your CMakeLists.txt file
<img width="673" alt="2" src="https://github.com/user-attachments/assets/f6638ecf-4a32-41a5-852c-3be0e232a4f1">
1.“changelog-file-not-compressed”: Compress changelog with gzip.This will generate changelog.gz, then i installed it in the CMake file.
<img width="919" alt="Screenshot 2024-12-09 at 2 37 12 PM" src="https://github.com/user-attachments/assets/bba622ec-82c0-44d5-adff-802bf9d5e28c">
